python/adelphi/tests/integration/__init__.py

```diff
@@ -0,0 +1,200 @@
+
+logging.basicConfig(filename="adelphi.log", level=logging.INFO)
+log = logging.getLogger('adelphi')
+
+TempDirs = namedtuple('TempDirs', 'basePath, outputDirPath')
+
+def digestSet(schemaFile):
+    rv = set()
+    for (_, digest) in cqlDigestGenerator(schemaFile):
+        rv.add(digest)
+    return rv
+
+
+def logCqlDigest(schemaFile, digestSet):
+    for (cql, digest) in cqlDigestGenerator(schemaFile):
+        if digest in digestSet:
+            log.info("Digest: {}, CQL: {}".format(digest,cql))
+
+
+class SchemaTestMixin:
+
+    def basePath(self, name):
+        return os.path.join(self.dirs.basePath, name)
+
+
+    def outputDirPath(self, name):
+        return os.path.join(self.dirs.outputDirPath, name)
+
+
+    def stdoutPath(self, version=None):
+        return self.basePath("{}-stdout.out".format(version))
+
+
+    def stderrPath(self, version=None):
+        return self.basePath("{}-stderr.log".format(version))
+
+
+    def makeTempDirs(self):
+        base = tempfile.mkdtemp()
+        outputDir = os.path.join(base, "outputDir")
+        os.mkdir(outputDir)
+        self.dirs = TempDirs(base, outputDir)
+
+
+    def connectToLocalCassandra(self):
+        session = None
+        while not session:
+            try:
+                cluster = Cluster(["127.0.0.1"],port=9042)
+                session = cluster.connect()
+
+                # Confirm that the session is actually functioning before calling things good
+                rs = session.execute("select * from system.local")
+                log.info("Connected to Cassandra cluster, first row of system.local: {}".format(rs.one()))
+                log.info("Cassandra cluster ready")
+                return (cluster,session)
+            except:
+                log.info("Couldn't quite connect yet, will retry")
+                time.sleep(1)
+
+
+    def createSchema(self, session=None):
+        schemaPath = self.getBaseSchemaPath()
+        log.info("Creating schema on Cassandra cluster from file {}".format(schemaPath))
+        with open(schemaPath) as schema:
+            buff = ""
+            for line in schema:
+                realLine = line.strip()
+                if len(realLine) == 0:
+                    log.debug("Skipping empty statement")
+                    continue
+                if realLine.startswith("//"):
+                    log.debug("Skipping commented statement {}".format(stmt))
+                    continue
+                buff += (" " if len(buff) > 0 else "")
+                buff += realLine
+                if realLine.endswith(';'):
+                    log.debug("Executing statement {}".format(buff))
+                    try:
+                        session.execute(buff)
+                    except:
+                        log.error("Exception executing statement: {}".format(buff), exc_info=sys.exc_info()[0])
+                        self.fail("Exception executing statement: {}, check log for details".format(buff))
```

Comment:
Failures to create the test schema will now fail the test.  This is more rigorous and seems more correct on it's face but note that we _were_ using this operation to silently handle the case of an inability to add SASI indexes to an instance which wasn't configured to support them.  Unsurprisingly that logic was also hiding _other_ schema failures (who could've seen that coming :facepalm: ) so it was yanked and the SASI index generation logic made conditional.